### DISCUSSION TITLE
Wondering how to run several different animations in sequence with React Native Reanimated.

### DISCUSSION THREAD

**Discussion by Yummy-sk** [software-mansion/react-native-reanimated#5294] [Q&A]
When running multiple different animations at once, I want to specify the order of execution between each animation.

For example, in React Native, a space like the one below.
- [Animated sequence](https://reactnative.dev/docs/animated#sequence)
- [Animated parallel](https://reactnative.dev/docs/animated#parallel)

Reanimated also has a [withSequence](https://docs.swmansion.com/react-native-reanimated/docs/animations/withSequence/), but it doesn't seem to be able to handle the order of execution of multiple animations.

So I'm wondering if there's a way to specify the order of execution of different animations, like sequence and parallel in React native.

### Animated Example
```javascript
const a = useRef(new Animated.Value(0))
const b = useRef(new Animated.Value(1))


const let animate = () => {
  Animated.sequence(
    a->Animated.config(...)
    b->Animated.config(...)
  ).start()
}
```

### Animated Example
```javascript
const a = useSharedValue(0)
const b = useSharedValue(1)


const let animate = () => {
  withSequence(
    a->withTiming(...)
    b->withTiming(...)
  ).start()

 // or
  withParallel(
    a->withTiming(...)
    b->withTiming(...)
  ).start()
}
```

**Comment by Noitidart** [comment]
Were you able to figure out `withParallel`?

**Comment by MatiPl01** [comment]
Hey!

Do you guys still have problems on reanimated v3? There is [withSequence](https://docs.swmansion.com/react-native-reanimated/docs/animations/withSequence/) documentation page, which seems to be what you are looking for.

In terms of `withParallel`, I wonder how it may be supposed to work. If you have multiple shared values and would like to animate them in parallel, you don't have to do any extra steps, just update multiple values one after another in such a way:

```tsx
const a = useSharedValue(0)
const b = useSharedValue(1)

// These both animations will run in parallel
a.value = withTiming(...)
b.value = withTiming(...)
```
